### PR TITLE
People 183 display skills

### PR DESCRIPTION
**Pull request by DariuszMusielak**
# Ticket
https://netguru.atlassian.net/browse/PEOPLE-183

# Description
- Added user_skill_rates policy
- Added partial for skills
- Added some stylesheets for skills on user profile page
- Added new method for query service for getting skills for display

### Before merge checklist
- [ ] CircleCI is passing
- [ ] Proper labels are set

screen shot 2017-02-23 at 10 55 45

### CODE REVIEW

File: app/views/users/show.html.haml [hunk 1]
```diff
@@ -6,6 +6,9 @@
           = user_show_page.user_gravatar
         = user_show_page.user.name
       = mail_to user.email, "send email"
+    .list-group
+      .list-group-item
+        = render 'skills_with_rating', grouped_skills: grouped_skills_by_category
```
Comment (netguru-hound) [inline app/views/users/show.html.haml]: Line is too long. [81/80]

File: app/views/users/_skills_with_rating.html.haml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        %span.label.skill-label{ class: skill_label_class(skill) }
+          - if skill.rate_type == 'range'
+            %span.badge
+              = skill.rate
+              - if skill.favorite
+                %i.glyphicon.glyphicon-heart.skill__icon
+          - elsif skill.favorite
+            %span.badge
+              %i.glyphicon.glyphicon-heart.skill__icon
+          = "#{skill.name}"
```
Comment (netguru-hound) [inline app/views/users/_skills_with_rating.html.haml]: `= "..."` should be rewritten as `...`

File: app/queries/user_skill_rates_query.rb [hunk 3]
```diff
@@ -9,6 +9,10 @@ def results_by_categories
     user_skill_rates.group_by(&:category).sort_by{ |key, _| key }.to_h
   end
 
+  def rated_skills
+    user_skill_rates.where('rate > 0').group_by(&:category).sort_by{ |key, _| key }.to_h
```
Comment (netguru-hound) [inline app/queries/user_skill_rates_query.rb]: Space missing to the left of {.

File: app/assets/stylesheets/controllers/users.scss [hunk 4]
```diff
@@ -7,7 +7,47 @@ body.users {
...
+      }
+      &--orange {
+        background-color: $color-sunset;
+        .badge {
+          color: $color-sunset;
+        }
+      }
+
+      .badge {
+        padding: 2px 3px;
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Properties should be ordered background-color, padding

File: app/assets/stylesheets/controllers/users.scss [hunk 5]
```diff
@@ -7,7 +7,47 @@ body.users {
...
+      }
+      &--yellow {
+        background-color: $color-yellow;
+        .badge {
+          color: $color-yellow;
+        }
+      }
+      &--orange {
+        background-color: $color-sunset;
+        .badge {
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Rule declaration should be preceded by an empty line<br>Nesting should be no greater than 4, but was 5

File: app/assets/stylesheets/controllers/users.scss [hunk 6]
```diff
@@ -7,7 +7,47 @@ body.users {
...
+          color: $color-green;
+        }
+      }
+      &--yellow {
+        background-color: $color-yellow;
+        .badge {
+          color: $color-yellow;
+        }
+      }
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Rule declaration should be followed by an empty line

File: app/assets/stylesheets/controllers/users.scss [hunk 7]
```diff
@@ -7,7 +7,47 @@ body.users {
...
+      display: inline-block;
+      &--green {
+        background-color: $color-green;
+        .badge {
+          color: $color-green;
+        }
+      }
+      &--yellow {
+        background-color: $color-yellow;
+        .badge {
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Rule declaration should be preceded by an empty line<br>Nesting should be no greater than 4, but was 5

File: app/assets/stylesheets/controllers/users.scss [hunk 8]
```diff
@@ -7,7 +7,47 @@ body.users {
...
+    .skill-label {
+      margin: 5px 5px 0 0;
+      padding-right: 7px;
+      display: inline-block;
+      &--green {
+        background-color: $color-green;
+        .badge {
+          color: $color-green;
+        }
+      }
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Rule declaration should be followed by an empty line

File: app/assets/stylesheets/controllers/users.scss [hunk 9]
```diff
@@ -7,7 +7,47 @@ body.users {
...
+    }
+
+    .skill-label {
+      margin: 5px 5px 0 0;
+      padding-right: 7px;
+      display: inline-block;
+      &--green {
+        background-color: $color-green;
+        .badge {
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Rule declaration should be preceded by an empty line<br>Nesting should be no greater than 4, but was 5

File: app/assets/stylesheets/controllers/users.scss [hunk 10]
```diff
@@ -7,7 +7,47 @@ body.users {
...
+        color: $color-red;
+        font-size: 10px;
+      }
+    }
+
+    .skill-label {
+      margin: 5px 5px 0 0;
+      padding-right: 7px;
+      display: inline-block;
+      &--green {
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Rule declaration should be preceded by an empty line

File: app/assets/stylesheets/controllers/users.scss [hunk 11]
```diff
@@ -7,7 +7,47 @@ body.users {
...
+        margin: 10px 0;
+      }
+      &__icon {
+        color: $color-red;
+        font-size: 10px;
+      }
+    }
+
+    .skill-label {
+      margin: 5px 5px 0 0;
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Properties should be ordered display, margin, padding-right<br>Shorthands of length `4` are not allowed. Value was `5px 5px 0 0`

File: app/assets/stylesheets/controllers/users.scss [hunk 12]
```diff
@@ -7,7 +7,47 @@ body.users {
       @extend .col-sm-12;
       padding-bottom: 4px;
     }
+
+    .skill {
+      &__hr {
+        margin: 10px 0;
+      }
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Rule declaration should be followed by an empty line

File: app/queries/user_skill_rates_query.rb [hunk 13]
```diff
@@ -6,7 +6,11 @@ def initialize(user)
   end
 
   def results_by_categories
-    user_skill_rates.group_by(&:category).sort_by{ |key, _| key }.to_h
+    user_skill_rates.group_by(&:category).sort_by { |key, _| key }.to_h
+  end
+
+  def rated_skills
+    user_skill_rates.where('rate > 0').group_by(&:category).sort_by{ |key, _| key }.to_h
```
Comment (netguru-hound) [inline app/queries/user_skill_rates_query.rb]: Space missing to the left of {.

File: app/assets/stylesheets/controllers/users.scss [hunk 14]
```diff
@@ -7,7 +7,50 @@ body.users {
...
+      &--yellow {
+        background-color: $color-yellow;
+        .badge {
+          color: $color-yellow;
+        }
+      }
+
+      &--orange {
+        background-color: $color-sunset;
+        .badge {
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Rule declaration should be preceded by an empty line<br>Nesting should be no greater than 4, but was 5

File: app/assets/stylesheets/controllers/users.scss [hunk 15]
```diff
@@ -7,7 +7,50 @@ body.users {
...
+      &--green {
+        background-color: $color-green;
+        .badge {
+          color: $color-green;
+        }
+      }
+
+      &--yellow {
+        background-color: $color-yellow;
+        .badge {
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Rule declaration should be preceded by an empty line<br>Nesting should be no greater than 4, but was 5

File: app/assets/stylesheets/controllers/users.scss [hunk 16]
```diff
@@ -7,7 +7,50 @@ body.users {
...
+    }
+
+    .skill-label {
+      margin: 5px 5px 0 0;
+      padding-right: 7px;
+      display: inline-block;
+
+      &--green {
+        background-color: $color-green;
+        .badge {
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Rule declaration should be preceded by an empty line<br>Nesting should be no greater than 4, but was 5

File: app/assets/stylesheets/controllers/users.scss (hunk 4 above)
Comment (marcinjak) [inline app/assets/stylesheets/controllers/users.scss]: :point_up:

File: app/assets/stylesheets/controllers/users.scss [hunk 17]
```diff
@@ -7,7 +7,53 @@ body.users {
...
+      &--orange {
+        background-color: $color-sunset;
+
+        .badge {
+          color: $color-sunset;
+        }
+      }
+
+      .badge {
+        padding: 2px 3px;
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Properties should be ordered background-color, padding

File: app/assets/stylesheets/controllers/users.scss [hunk 18]
```diff
@@ -7,7 +7,53 @@ body.users {
...
+
+        .badge {
+          color: $color-yellow;
+        }
+      }
+
+      &--orange {
+        background-color: $color-sunset;
+
+        .badge {
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Nesting should be no greater than 4, but was 5

File: app/assets/stylesheets/controllers/users.scss [hunk 19]
```diff
@@ -7,7 +7,53 @@ body.users {
...
+          color: $color-green;
+        }
+      }
+
+      &--yellow {
+        background-color: $color-yellow;
+
+        .badge {
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Nesting should be no greater than 4, but was 5

File: app/assets/stylesheets/controllers/users.scss [hunk 20]
```diff
@@ -7,7 +7,53 @@ body.users {
...
+
+    .skill-label {
+      margin: 5px 5px 0 0;
+      padding-right: 7px;
+      display: inline-block;
+
+      &--green {
+        background-color: $color-green;
+
+        .badge {
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Nesting should be no greater than 4, but was 5

File: app/assets/stylesheets/controllers/users.scss (hunk 17 above)
Comment (marcinjak) [inline app/assets/stylesheets/controllers/users.scss]: :point_up:

File: spec/queries/user_skill_rates_query_spec.rb [hunk 21]
```diff
@@ -8,8 +8,22 @@
...
   let(:skill_frontend) { create(:skill, skill_category: skill_category_frontend) }
-  let!(:user_skill_rate_backend) { create(:user_skill_rate, user: user, skill: skill_backend ) }
-  let!(:user_skill_rate_frontend) { create(:user_skill_rate, user: user, skill: skill_frontend ) }
+  let!(:user_skill_rate_backend) do
+    create(
+      :user_skill_rate,
+      user: user,
+      skill: skill_backend,
+      rate: 1
+      )
```
Comment (marcinjak) [inline spec/queries/user_skill_rates_query_spec.rb]: :point_up:

File: spec/queries/user_skill_rates_query_spec.rb [hunk 22]
```diff
@@ -67,4 +81,13 @@
       )
     end
   end
+
+  describe '#rated_skills' do
+    let(:expected_results){ { "backend" => [user_skill_rate_backend] } }
```
Comment (marcinjak) [inline spec/queries/user_skill_rates_query_spec.rb]: :point_up:

File: app/assets/stylesheets/controllers/users.scss (hunk 17 above)
Comment (DariuszMusielak) [inline app/assets/stylesheets/controllers/users.scss]: According to this: https://netguru.atlassian.net/wiki/display/FRONT/CSS+Styleguide#CSSStyleguide-Order background-color should be at the end.

File: app/assets/stylesheets/controllers/users.scss [hunk 23]
```diff
@@ -7,7 +7,55 @@ body.users {
...
+      &--orange {
+        background-color: $color-sunset;
+
+        .badge {
+          color: $color-sunset;
+        }
+      }
+
+      .badge {
+        padding: 2px 3px;
```
Comment (netguru-hound) [inline app/assets/stylesheets/controllers/users.scss]: Properties should be ordered background-color, line-height, padding

**Pull request merged**